File: apps/starter-ios/StarterApp/Core/Billing/BillingModels.swift
import Foundation

struct VerifyNativePurchaseBody: Encodable {
    let platform: String
    let productId: String
    let purchaseToken: String
    let transactionId: String?
}

struct EntitlementState: Codable, Equatable {
    let tier: String
    let source: String
    let features: [String]
    let renewsAt: String?
}

struct VerifyNativePurchaseResponse: Codable, Equatable {
    let verified: Bool
    let productId: String
    let entitlement: EntitlementState
}

File: apps/starter-android/app/src/main/kotlin/com/onlystack/starterapp/core/billing/BillingModels.kt
package com.onlystack.starterapp.core.billing

data class EntitlementState(
    val tier: String,
    val source: String,
    val features: List<String>,
    val renewsAt: String? = null,
)

data class VerifyNativePurchaseBody(
    val platform: String,
    val productId: String,
    val purchaseToken: String,
    val transactionId: String? = null,
)

data class VerifyNativePurchaseResponse(
    val verified: Boolean,
    val productId: String,
    val entitlement: EntitlementState,
)
